ffi/diplomat/src/locale.rs

```diff
@@ -126,7 +114,7 @@ pub mod ffi {
                     self.0.id.language = language;
                     Ok(())
                 }
-                Err(_) => Err(ICU4XLocaleError::Error),
+                Err(_) => Err(ICU4XError::LocaleParserError),
```

Comment:
Thought: This seems like the kind of error that maybe should be more generic. In ICU4X, this is an `Option`, and basically we're returning an error when the `Option` is `None`. Maybe `OptionNoneError` or `MissingOptionError`?

Reply:
I don't understand what you mean by it being an option; it's a ParserError in locid?

Reply:
Sorry, wrote the comment on the wrong line. My comment applies to the getter functions like `.region()` and `.script()` that return an Option in Rust.

Reply:
Ah! I think part of the issue is that returning None writeables isn't a thing, and I'm not sure if it shoudl be because empty vs null becomes a subtle distinction that javascript doesn't like too much. Perhaps we should have that error, yeah.